magento/magento2
# Issue-36045: Category Sort (position, name, price) behavior has changed

### Description (*)
Category Sort (position, name, price) behavior has changed in M2.4.5

The sorted order of products appearing in catalog category pages is controlled by the Sort By setting selected by the user from a drop down appearing in the upper right corner of the category page.

Previously the out-of-stock status of a product had no effect on the sort order, but as of update to M2.4.5 this appears to be working differently, out-of-stock items are not sorting in to the correct order regardless of setting. Instead, starting with M2.4.5 the out-of-stock items are always listed at the end.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#36045

### Manual testing scenarios (*)

1. perform clean installtion of M2.4.5 with the Luma store data
2. In frontend website navigate to Home > Gear > Watches, and observe that the position of Luma Analog Watch is appropriate and correct in the order of products displayed for all Sort By selections.
3. In admin backend Catalog > Product > Luma Analog Watch > Advanced Inventory, uncheck Use Config Settings and change Out-of-Stock Threshold value to be greater than current Qty, in this case value = 101, alternately change the current QTY value to zero (either method will cause same results) then Done and Save.
4. in admin backend Stores > Configuration > Catalog > Inventory, uncheck Use System Value and change Display Out Of Stock Products value to Yes, then Save Config
5. in admin backend Cache Management, Flush Magento Cache
6. In frontend Home > Gear > Watches observe that the position of Luma Analog Watch is ALWAYS at the end of products displayed regardless of Sort By selection
7. In admin Stores > Configuration > Catalog > Inventory, uncheck Use System Value and change Display Out Of Stock Products At The End Of The Catalog value to Yes, then Save Config
8. in admin backend Cache Management, Flush Magento Cache
9. Navigate to Home > Gear > Watches, check position Luma Analog Watch in frontend

### Questions or comments


### Contribution checklist (*)
